### PR TITLE
Edit feature template

### PR DESCRIPTION
**Pull request by cicharka**
# Pull Request summary:
[Describe briefly what the PR is about]

# Description of changes:
[Add more in depth analysis of what changed, provide logs, examples of usage]

# Checklist:
- [x] Make sure to run pre-commit before commiting changes
- [x] Make sure all checks have passed
- [ ] PR description is clear and comprehensive
- [ ] Mentioned the issue that this PR solves (if applicable)
- [x] Make sure you test the changes

### CODE REVIEW

File: vmngclient/api/templates/feature_template.py [hunk 1]
```diff
@@ -45,3 +46,30 @@ def payload_path(self) -> Path:
...
+        """Gets feature template model corresponding to existing feature template based on provided name
+
+        Args:
+            session: vManageSession
+            name: name of the existing feature template
+
+        Returns:
+            FeatureTemplate: filed out feature template model
+        """
+        from vmngclient.utils.feature_template import choose_model, find_template_values
```
Comment (tehAgitto) [inline vmngclient/api/templates/feature_template.py:61]: move import on the beginning of the file

File: vmngclient/api/templates/models/supported.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+from vmngclient.api.templates.models.cisco_aaa_model import CiscoAAAModel
+from vmngclient.dataclasses import AlarmData
+
+
+class FeatureTemplateType(Enum):
+    cisco_aaa: str = "cedge_aaa"
+
+
+supported_models = [CiscoAAAModel, AlarmData]  # AlarmData provided just for mypy (>1 class needed, only 1 model)
```
Comment (tehAgitto) [inline vmngclient/api/templates/models/supported.py:11]: I think Security Template is also supported.

**Review by tehAgitto — CHANGES_REQUESTED** [review]

File: vmngclient/api/templates/feature_template.py (hunk 1 above)
Comment (tomaszwarda) [reply vmngclient/api/templates/feature_template.py:61]: cannot do this, circular import created

File: vmngclient/api/templates/models/supported.py (hunk 2 above)
Comment (tomaszwarda) [reply vmngclient/api/templates/models/supported.py:11]: added just for wip, after rebase will be fixed

**Review by tehAgitto — APPROVED** [review]

File: vmngclient/utils/feature_template.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
+from typing import Callable, Dict, Optional, Union
+
+from vmngclient.api.templates.models.supported import FeatureTemplateType, supported_models
+from vmngclient.exceptions import TemplateTypeError
+
+
+def choose_model(type_value: Union[str, FeatureTemplateType]) -> Callable:
```
Comment (sbasan) [inline vmngclient/utils/feature_template.py:7]: If we want to pick a model class by template name identifier why don't use a dict? (see my comment supported.py)
Also return type seems wrong.

File: vmngclient/dataclasses.py [hunk 4]
```diff
@@ -259,6 +261,34 @@ class User(DataclassBase):
...
+@define
+class TemplateField(DataclassBase):
+    name: str
+    definition: Optional[TemplateFieldDefinition] = field(default=None)
+    template_dicts: Optional[TemplateField] = field(default=None)
+
+    @definition.validator  # type: ignore
+    def definition_is_valid(self, attribute, value):
+        if value is not None:
+            assert self.template_dicts is None  # if definition is provided, cannot have template_dicts at the same time
```
Comment (sbasan) [inline vmngclient/dataclasses.py:284]: Maybe use ValueError("'definition' and 'template_dicts' are mutually exclusive.") ?

File: vmngclient/api/templates/feature_template.py (hunk 1 above)
Comment (sbasan) [reply vmngclient/api/templates/feature_template.py:61]: I would try to find a design in which there are no circular buffers, maybe modules could be arranged better?

Comment (sbasan) [reply vmngclient/api/templates/feature_template.py:61]: I would try to find a design in which there are no circular imports, maybe modules could be arranged better?

**Review by sbasan — CHANGES_REQUESTED** [review]

**Comment by tehAgitto** [comment]
Closes #219

File: vmngclient/dataclasses.py (hunk 4 above)
Comment (tehAgitto) [reply vmngclient/dataclasses.py]: Dataclass is removed.

**Pull request merged**